### PR TITLE
VRR Client

### LINKED ISSUE #172: Create Client for VRR-API
```[tasklist]
### Tasks
```

### PR DESCRIPTION
**Pull request by Paulomart**
### Description

This is the Pull Request Template for the Oeffis Project

## Checklist

- [ ] My code follows the style guidelines
- [ ] I have documented my code
- [ ] My changes generate no new warnings
- [ ] I have linked the Issue
- [ ] The Pull Request covers all the Checkpoints from the tagged Issue
- [ ] I have noted the hours of work i have put in the issue

## I have edited following Parts of the project

- [ ] Frontend
- [ ] Backend
- [ ] Github
- [ ] Root

## If tests are required did you make them?

- [ ] No tests required
- [ ] Yes
- [ ] No

Closes #172

### CODE REVIEW

**Comment by Paulomart** [comment]
@timvahlbrock We need a concept on how to test this client. Got some ideas?

**Comment by timvahlbrock** [comment]
I think this will mainly depend on the interface between the backend and this client. In general, cypress is pretty versatile, so we might be able to use the existing e2e infrastructure.

File: vrr_client/src/test.ts
Comment (timvahlbrock) [inline vrr_client/src/test.ts:1]: Do you really want to merge this file?

File: vrr_client/src/Constants.ts [hunk 1]
```diff
@@ -0,0 +1 @@
+export const VRR_TEST_API_BASE_URL = "https://openservice-test.vrr.de/";
```
Comment (timvahlbrock) [inline vrr_client/src/Constants.ts:1]: As discussed, we need to be able to override this URL during testing.

File: vrr_client/src/StopFinderClient.ts
Comment (timvahlbrock) [inline vrr_client/src/StopFinderClient.ts:1]: In reality, we would probably rather calculate a route from coordinates, not from the closest station, as the later might not actually be the fastest route. But maybe this is fine for now?

**Review by timvahlbrock — CHANGES_REQUESTED** [review]

File: backend/package.json [hunk 2]
```diff
@@ -23,7 +23,8 @@
     "@nestjs/swagger": "^6.3.0",
     "hafas-client": "^6.0.5",
     "reflect-metadata": "^0.1.13",
-    "rxjs": "^7.2.0"
+    "rxjs": "^7.2.0",
+    "@oeffis/vrr_client": "file:../vrr_client"
```
Comment (timvahlbrock) [inline backend/package.json:27]: You have included the var client, but it's not actually used in src.

File: vrr_client/src/Constants.ts (hunk 1 above)
Comment (Paulomart) [reply vrr_client/src/Constants.ts:1]: You can provide any url in the constructor of the client.

File: backend/package.json (hunk 2 above)
Comment (Paulomart) [reply backend/package.json:27]: Yes thats by design to validate the dependency tree.

File: vrr_client/src/StopFinderClient.ts
Comment (Paulomart) [reply vrr_client/src/StopFinderClient.ts:1]: We can do both by using this api.

File: vrr_client/src/test.ts
Comment (Paulomart) [reply vrr_client/src/test.ts:1]: I will move it under a different name + folder

File: vrr_client/src/Constants.ts (hunk 1 above)
Comment (timvahlbrock) [reply vrr_client/src/Constants.ts:1]: for E2E test we need to modify it using env vars. but I can implement this in my pr.

**Review by timvahlbrock — APPROVED** [review]

File: vrr_client/src/StopFinderClient.ts
Comment (timvahlbrock) [reply vrr_client/src/StopFinderClient.ts:1]: what I mean is: the closest station might not go into the direction I wanna go to. maybe a station, that is 1 minute further away has options that take me to my destination much faster. if we only consider the station that is closes to ourselves, we won't see those.

**Pull request merged**